src/nft/components/sell/list/ListPage.tsx

```diff
@@ -1,3 +1,841 @@
+              LAST: {formatEth(asset.lastPrice)} ETH
+            </Row>
+            <Row display={asset.floorPrice ? 'flex' : 'none'}>FLOOR: {formatEth(asset.floorPrice)} ETH</Row>
+          </>
+        ) : (
+          <>
+            {warning
+              ? warning.message
+              : warningType !== WarningType.NONE && (
+                  <>
+                    {warningType}
+                    {warningType === WarningType.BELOW_FLOOR
+                      ? formatEth(asset.floorPrice)
+                      : formatEth(asset.floor_sell_order_price)}
+                    ETH
+                    <Box
+                      color={warningType === WarningType.BELOW_FLOOR ? 'genieBlue' : 'orange'}
+                      marginLeft="8"
+                      cursor="pointer"
+                      onClick={() => {
+                        warningType === WarningType.ALREADY_LISTED && removeSellAsset(asset)
+                        setWarningType(WarningType.NONE)
+                      }}
+                    >
+                      {warningType === WarningType.BELOW_FLOOR ? 'DISMISS' : 'REMOVE ITEM'}
+                    </Box>
+                  </>
+                )}
+          </>
+        )}
+      </Row>
+    </Column>
+  )
+}
+
+export const formatEth = (price: number) => {
+  if (price > 1000000) {
+    return `${Math.round(price / 1000000)}M`
+  } else if (price > 1000) {
+    return `${Math.round(price / 1000)}K`
+  } else {
+    return `${Math.round(price * 100 + Number.EPSILON) / 100}`
+  }
+}
+
+const EthPriceDisplay = ({ ethPrice }: { ethPrice: number }) => {
+  const nanCheckedPrice = isNaN(ethPrice) ? 0 : ethPrice
+  const [ethConversion, setEthConversion] = useState(3000)
+  useEffect(() => {
+    fetchPrice().then((price) => {
+      setEthConversion(price ?? 0)
+    })
+  }, [])
+  return (
+    <Column width="full">
+      <Row width="full" justifyContent="flex-end">
+        <Box className={subhead} color={nanCheckedPrice !== 0 ? 'blackBlue' : 'darkGray'} marginLeft="auto">
+          {formatEth(nanCheckedPrice)} ETH
+        </Box>
+        {nanCheckedPrice !== 0 && (
+          <Box className={body} color="darkGray" marginLeft="12" marginRight="0">
+            {formatUsdPrice(nanCheckedPrice * ethConversion)}
+          </Box>
+        )}
+      </Row>
+    </Column>
+  )
+}
+
+export interface NFTListRowProps {
+  asset: WalletAsset
+  globalPriceMethod?: SetPriceMethod
+  setGlobalPrice: Dispatch<number>
+  globalPrice?: number
+  selectedMarkets: ListingMarket[]
+}
+
+function maxMarketFee(markets: ListingMarket[]): number {
+  let max = -1
+  markets.forEach((market) => {
+    if (market.fee > max) {
+      max = market.fee
+    }
+  })
+  return max
+}
+
+const MarketplaceRow = ({
+  globalPriceMethod,
+  globalPrice,
+  setGlobalPrice,
+  selectedMarkets,
+  removeMarket = undefined,
+  asset,
+  showMarketplaceLogo,
+}: any) => {
+  const [listPrice, setListPrice] = useState<string>()
```

Comment:
`string | undefined` if no default value

Reply:
this constructor automatically adds the undefined type to the param
![Screen Shot 2022-09-21 at 10 55 30 ](https://user-images.githubusercontent.com/11512321/191576445-c104c590-5922-483b-b3f7-5a52238ba1a1.png)